### ISSUE TITLE
[FEAT EXTRACTION] Does the chosen interpolator apply to both the image and mask?

### ISSUE THREAD

**Issue by mattwarkentin** [Radiomics/pyradiomics#595]
Hi,

I think this should be a pretty straightforward question. I know you can specify an interpolator in the YAML configuration file with:
```yaml
# Example
setting:
  interpolator: 'sitkBspline'
  resampledPixelSpacing: [1, 1, 1]
```
I am under the impression that this does resampling for both the image and mask. However, I would expect that the mask is resampled using nearest-neighbours interpolator. I note this is how this setting is described in the documentation [here](https://pyradiomics.readthedocs.io/en/latest/customization.html#feature-extractor-level)
> `interpolator` [sitkBSpline]: SimpleITK constant or string name thereof, sets interpolator to use for resampling. Enumerated value, possible values:

Further, one of the example files found [here](https://github.com/Radiomics/pyradiomics/blob/master/examples/exampleSettings/exampleCT.yaml) suggests that if you resample with the above settings:
```yaml
# correctMask and geometryTolerance are not needed, as both image and mask are resampled
```
So I am not sure if the same `interpolator` is used for both the image and mask, or whether NN is used for the mask. Just looking for clarification. Thanks for your time.

**Comment by JoostJM** [comment]
Mask resampling is forced to nearest neighbor, as we support other label values than 1. If you were to apply a different interpolation algorithm and rounding to integers, or thresholding, you risk changing label values if the label is not 1. Therefore we opted for nearest neighbor, which does not change the label value.

**Comment by mattwarkentin** [comment]
@JoostJM Thanks for the response. I understand the logic behind using NN for mask resampling, but I still think the documentation and example files do not make it explicitly clear, and perhaps leave the user under suspicion of whether the `interpolator` specified in the YAML file is applied to both the image and mask (which it is not).

Perhaps the documentation [here](https://pyradiomics.readthedocs.io/en/latest/customization.html#feature-extractor-level) should be updated in some way, such as:
> `interpolator` [sitkBSpline]: SimpleITK constant or string name thereof, sets interpolator to use for resampling. __The choice of `interpolator` is only applied to resampling images, while `sitkNearestNeighbor` is always used for resampling masks in order to preserve label values.__

Do you agree?

**Comment by JoostJM** [comment]
Certainly! Are you able to propose this change in a pull request? The documentation is automatically built from source code and additional files in the docs folder in the pyradiomics repository. The customization section is contained in `docs/customization.rst`.

**Comment by mattwarkentin** [comment]
Sure, I will submit a PR suggesting this update to the documentation.

**Comment by mattwarkentin** [comment]
Closed with #599
